src/ua/artcode/udiary/tests/TestMainController.java

```diff
@@ -0,0 +1,321 @@
+package ua.artcode.udiary.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import ua.artcode.udiary.controller.MainController;
+import ua.artcode.udiary.controller.MainControllerImpl;
+import ua.artcode.udiary.dao.AppDataContainer;
+import ua.artcode.udiary.dao.DairyDaoImpl;
+import ua.artcode.udiary.dao.RecordDaoImpl;
+import ua.artcode.udiary.dao.UserDaoImpl;
+import ua.artcode.udiary.exception.AppException;
+import ua.artcode.udiary.model.*;
+
+public class TestMainController {
+
+    private MainController mainController;
+
+    @Before
+    public void prepareTest() {
+        AppDataContainer appDataContainer = new AppDataContainer();
+        mainController = new MainControllerImpl(
+                new RecordDaoImpl(appDataContainer),
+                new DairyDaoImpl(appDataContainer),
+                new UserDaoImpl(appDataContainer)
+        );
+    }
+
+    @Test
+    public void addUser() {
+        // must not catch exception:
+        try {
+            mainController.addUser(new User("abc", "111"));
+        } catch (AppException e) {
+            Assert.fail("was caught AppException");
```

Comment:
we should log exception 'e'. at least do printstacktrace

Reply:
I've added it. You can check it in the next push.